Orange/tests/test_table.py

```diff
@@ -53,285 +54,146 @@ def test_filename(self):
 
-            # regular, discrete
-            varc = d.domain["c"]
-            self.assertEqual(d[0, 1], "0")
-            self.assertEqual(d[0, varc], "0")
-            self.assertEqual(d[0, "c"], "0")
-            self.assertEqual(d[0][1], "0")
-            self.assertEqual(d[0][varc], "0")
-            self.assertEqual(d[0]["c"], "0")
-            self.assertEqual(d[np.int_(0), np.int_(1)], "0")
-            self.assertEqual(d[np.int_(0)][np.int_(1)], "0")
-
-            # regular, continuous
-            varb = d.domain["b"]
-            self.assertEqual(d[0, 0], 0)
-            self.assertEqual(d[0, varb], 0)
-            self.assertEqual(d[0, "b"], 0)
-            self.assertEqual(d[0][0], 0)
-            self.assertEqual(d[0][varb], 0)
-            self.assertEqual(d[0]["b"], 0)
-            self.assertEqual(d[np.int_(0), np.int_(0)], 0)
-            self.assertEqual(d[np.int_(0)][np.int_(0)], 0)
-
-            # negative
-            varb = d.domain["b"]
-            self.assertEqual(d[-2, 0], 3.333)
-            self.assertEqual(d[-2, varb], 3.333)
-            self.assertEqual(d[-2, "b"], 3.333)
-            self.assertEqual(d[-2][0], 3.333)
-            self.assertEqual(d[-2][varb], 3.333)
-            self.assertEqual(d[-2]["b"], 3.333)
-            self.assertEqual(d[np.int_(-2), np.int_(0)], 3.333)
-            self.assertEqual(d[np.int_(-2)][np.int_(0)], 3.333)
-
-            # meta, discrete
-            vara = d.domain["a"]
-            metaa = d.domain.index("a")
-            self.assertEqual(d[0, metaa], "A")
-            self.assertEqual(d[0, vara], "A")
-            self.assertEqual(d[0, "a"], "A")
-            self.assertEqual(d[0][metaa], "A")
-            self.assertEqual(d[0][vara], "A")
-            self.assertEqual(d[0]["a"], "A")
-            self.assertEqual(d[np.int_(0), np.int_(metaa)], "A")
-            self.assertEqual(d[np.int_(0)][np.int_(metaa)], "A")
-
-            # meta, string
-            vare = d.domain["e"]
-            metae = d.domain.index("e")
-            self.assertEqual(d[0, metae], "i")
-            self.assertEqual(d[0, vare], "i")
-            self.assertEqual(d[0, "e"], "i")
-            self.assertEqual(d[0][metae], "i")
-            self.assertEqual(d[0][vare], "i")
-            self.assertEqual(d[0]["e"], "i")
-            self.assertEqual(d[np.int_(0), np.int_(metae)], "i")
-            self.assertEqual(d[np.int_(0)][np.int_(metae)], "i")
+            for dom, indexOne, indexTwo, comp in (("c", 0, 1, "0"),  # regular, discrete
+                                                  ("b", 0, 0, 0),  # regular, continuous
+                                                  ("b", -2, 0, 3.333),  # negative
+                                                  ("a", 0, 0, "A"),  # meta, discrete
+                                                  ("e", 0, 0, "i")):  # meta, string
+                if dom == "a" or dom == "e":
+                    indexTwo = d.domain.index(dom)
```

Comment:
This looks pretty weird at first glance. Perhaps you can use this second type of index in all iterations (regardless of how it was before only in the last two). Like so:
```py
var_domain_index = d.domain.index(var)
self.assertEqual(d[inst, var_domain_index], value)
...
```